### ISSUE TITLE
Guild events calendar sync

### ISSUE THREAD

**Issue by ennioVisco** [DreamExposure/DisCal-Discord-Bot#126]
** Current limitation **
Right the bot does not consider the server/guild scheduled events

** Nice to have **
It would be nice to have the bot do 2-way syncing with the calendar

**Additional context**
Scheduled events API seem to be available: https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event

**Comment by smnc** [comment]
This is actually already on the roadmap for a future version of the bot, but doesn't have an open issue yet.
I'm not 100% sure on its current status, but I know @NovaFox161 is working on it.

**Comment by NovaFox161** [comment]
I would absolutely love to have this supported in some form, but I'm waiting on the library we use to implement it (Discord4J)

The problem is, I'm the one working on the PR for this (Discord4J/Discord4J#1046), but the discord API for this is fairly messy and non-ideal, so I'm not 100% how to actually implement it into D4J (mainly the caching portions).

Until I or another person finishes the PRs I started, the feature is stalled on DisCal unfortunately. But, I am actively working on building support for several new things in the bot so development isn't waiting on this one feature.

**Comment by ennioVisco** [comment]
> I would absolutely love to have this supported in some form, but I'm waiting on the library we use to implement it (Discord4J)
> 
> The problem is, I'm the one working on the PR for this ([Discord4J/Discord4J#1046](https://github.com/Discord4J/Discord4J/pull/1046)), but the discord API for this is fairly messy and non-ideal, so I'm not 100% how to actually implement it into D4J (mainly the caching portions).
> 
> Until I or another person finishes the PRs I started, the feature is stalled on DisCal unfortunately. But, I am actively working on building support for several new things in the bot so development isn't waiting on this one feature.

I have some free time and some decent Java developing experience, what issue can I support you with? 
It's the first time I've seen the Discord4j library...

Also, the API from the docs here: https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event
looks fairly straightforward; what do you find to be messy? Is it maybe different than the other older ones?

**Comment by NovaFox161** [comment]
I appreciate the thought, but you fundamentally misunderstand what the issue is.

**Comment by DodoLeDev** [comment]
Any news on this? I've seen that Guild Events [seems to be merged on upstream Discord4J](https://github.com/Discord4J/Discord4J/pull/1134)...

That's the only thing holding me back from inviting it to my server!

**Comment by NovaFox161** [comment]
Haven't gotten to it yet due to life and work. Right now just doing maintenance work on the bot till I have time to work on a major feature addition (the maintenance work also makes it easier to add features in the future).
